browser/ui/webui/chat_ui/chat_ui_page_handler.cc

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ui/webui/chat_ui/chat_ui_page_handler.h"
+
+#include <algorithm>
+#include <utility>
+#include <vector>
+
+#include "base/strings/strcat.h"
+#include "brave/components/chat_ui/common/chat_ui_constants.h"
+#include "chrome/browser/ui/tabs/tab_strip_model.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/storage_partition.h"
+
+ChatUIPageHandler::ChatUIPageHandler(
+    TabStripModel* tab_strip_model,
+    mojo::PendingReceiver<chat_ui::mojom::PageHandler> receiver)
+    : receiver_(this, std::move(receiver)) {
+  DCHECK(tab_strip_model);
+  tab_strip_model->AddObserver(this);
```

Comment:
Please take a look at BrowserTabStripTracker, It seems this is a more accurate way to observe TSMs.